### PR TITLE
Added new Google Analytics events

### PR DESCRIPTION
**Pull request by mikekeehnen**

### CODE REVIEW

File: packages/googleanalytics/events/gtagAddToCart/GtagAddToCart.graphql [hunk 1]
```diff
@@ -1,5 +1,11 @@
 fragment GtagAddToCart on Cart @inject(into: ["CartItemCountChanged"]) {
   items {
+    product{
+      manufacturer
```
Comment (paales) [inline packages/googleanalytics/events/gtagAddToCart/GtagAddToCart.graphql:4]: `manufacturer` isn't a system attribute, so isn't allowed to be used.

**Review by paales — CHANGES_REQUESTED** [review]

File: packages/googleanalytics/events/gtagAddToCart/GtagAddToCart.graphql (hunk 1 above)
Comment (mikekeehnen) [reply packages/googleanalytics/events/gtagAddToCart/GtagAddToCart.graphql:4]: Do you know a way to get the brand of a product as a system attribute?

File: packages/googleanalytics/plugins/GaViewItem.tsx [hunk 2]
```diff
@@ -10,7 +10,12 @@ export const exported = '@graphcommerce/magento-product'
 function GaViewItem(props: PluginProps<ProductPageMetaFragment>) {
   const { Prev, url_key, ...rest } = props
 
-  useEffect(() => gtagViewItem(rest), [rest])
+  useEffect(
+    () => gtagViewItem(rest),
+    // We're disabling eslint rule to prevent this event from being triggerd on every rerender
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
+  )
```
Comment (paales) [inline packages/googleanalytics/plugins/GaViewItem.tsx:18]: What if we navigate from a product page to another product page, that probably wont retrigger here, right?

File: packages/googleanalytics/package.json [hunk 3]
```diff
@@ -12,14 +12,14 @@
...
-    "@graphcommerce/next-ui": "5.2.0-canary.17",
-    "@graphcommerce/magento-cart-items": "5.2.0-canary.17"
+    "@graphcommerce/graphql-mesh": "6.0.0-canary.24",
+    "@graphcommerce/magento-cart": "6.0.0-canary.24",
+    "@graphcommerce/magento-cart-payment-method": "6.0.0-canary.24",
+    "@graphcommerce/magento-cart-shipping-method": "6.0.0-canary.24",
+    "@graphcommerce/magento-product": "6.0.0-canary.24",
+    "@graphcommerce/next-config": "6.0.0-canary.24",
+    "@graphcommerce/next-ui": "6.0.0-canary.24",
+    "@graphcommerce/magento-cart-items": "6.0.0-canary.24"
```
Comment (paales) [inline packages/googleanalytics/package.json:22]: I think rebasing should do the trick, then this shouldn't be here.

File: packagesDev/graphql-codegen-relay-optimizer-plugin/dist/module/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { PluginFunction } from '@graphql-codegen/plugin-helpers';
+export interface RelayOptimizerPluginConfig {
+}
+export declare const plugin: PluginFunction<RelayOptimizerPluginConfig>;
```
Comment (paales) [inline packagesDev/graphql-codegen-relay-optimizer-plugin/dist/module/index.d.ts:4]: I think rebasing should do the trick, then this shouldn't be here.

**Review by paales — CHANGES_REQUESTED** [review]

File: packages/googleanalytics/events/gtagAddToCart/GtagAddToCart.graphql (hunk 1 above)
Comment (paales) [reply packages/googleanalytics/events/gtagAddToCart/GtagAddToCart.graphql:4]: Brand isn't a system attribute unfortunately :(

All values are prefixed with `item_`, maybe allow users to inject something like this that the user can add locally?

```
fragment MyAttr on CartItem @inject(into: ["CartItemCountChanged"]) {
   product {
      item_brand: manufacturere
   }
}
```

File: packages/googleanalytics/plugins/GaViewItem.tsx (hunk 2 above)
Comment (paales) [reply packages/googleanalytics/plugins/GaViewItem.tsx:18]: Maybe we can do something like:

```tsx
const viewItem = useMemoDeep(rest);
useEffect(() => {}, [viewItem])
```

**Review by paales — CHANGES_REQUESTED** [review]

File: packages/googleanalytics/plugins/GaCartStartCheckoutLinkOrButton.tsx [hunk 5]
```diff
@@ -10,6 +12,18 @@ export function GaCartStartCheckoutLinkOrButton(
...
+  useEffect(
+    () =>
+      gtagViewCart({
+        items: rest.items,
+        prices: rest.prices,
+      }),
+    // We're disabling eslint rule to prevent this event from being triggerd on every rerender
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
+  )
```
Comment (paales) [inline packages/googleanalytics/plugins/GaCartStartCheckoutLinkOrButton.tsx:25]: Should be replaced with useMemoObject (previously useStableObject)

**Pull request merged**